R/formula.R

```diff
@@ -0,0 +1,283 @@
+#' @title JASP Formulas
+#'
+#' @description These functions provide support to [stats::formula] in R syntax. They are used to internally parse formula objects.
+#' These functions are not intended for direct use.
```

Comment:
Great that you added documentation! The one thing I wonder about is whether the current name `createJaspFormula` is not confusing for module developers, who use `createJaspXXX` all the time. They might think, "oh is this something we can use when writing a module?"
You probably though the same, hence the sentence you wrote. Nevertheless, I feel we could explain a bit more. How about:
```suggestion
#' These functions are **not** intended for direct use when running JASP syntax in R or when developing JASP modules. They are used indirectly by JASP modules to validate their input formulas.
```
provided the \*\*not\*\* renders properly.

Reply:
Good point. I did not even realize the connection to the other `createJasp*` functions.

I did what you suggested + renamed the function to `makeJaspFormula` + added `@keywords internal` to the documentation to make it clearer this is intended for internal use only